Makefile

```diff
@@ -1,9 +1,40 @@
+PROJECT := liftbridge-api
+
+BUF_VERSION := 0.31.1
+
+UNAME_OS := $(shell uname -s)
+UNAME_ARCH := $(shell uname -m)
+CACHE_BASE := $(HOME)/.cache/$(PROJECT)
+CACHE := $(CACHE_BASE)/$(UNAME_OS)/$(UNAME_ARCH)
+CACHE_BIN := $(CACHE)/bin
+CACHE_VERSIONS := $(CACHE)/versions
+
+export PATH := $(abspath $(CACHE_BIN)):$(PATH)
+
+BUF := $(CACHE_VERSIONS)/buf/$(BUF_VERSION)
+$(BUF):
+	@rm -f $(CACHE_BIN)/buf
+	@mkdir -p $(CACHE_BIN)
+	curl -sSL \
+		"https://github.com/bufbuild/buf/releases/download/v$(BUF_VERSION)/buf-$(UNAME_OS)-$(UNAME_ARCH)" \
+		-o "$(CACHE_BIN)/buf"
+	chmod +x "$(CACHE_BIN)/buf"
+	@rm -rf $(dir $(BUF))
+	@mkdir -p $(dir $(BUF))
+	@touch $(BUF)
+
 .PHONY: all
 all: go py
 
 .PHONY: go
 go:
 	protoc --gofast_out=plugins=grpc:go api.proto
```

Comment:
Just a note - you could install `protoc` with the same methodology as above and make it so that `protoc` is upgraded by changing the `PROTOC_VERSION` variable as well https://github.com/bufbuild/makego/blob/master/make/go/dep_protoc.mk#L12

You could also remove `protoc` entirely here and just use [buf generate](https://docs.buf.build/generate-usage/).

```yaml
# buf.gen.go
version: v1beta1
plugins:
  - name: gofast
    out: go
    opt: plugins=grpc
```

Reply:
Thank you, that's useful. If @tylertreat is interested, I could work on the generation bits in a new PR. @bufdev, for Python, can `buf generate` be used or we still need `python -m grpc_tools.protoc`?

Reply:
I'm not sure on how `python -m grpc_tools.protoc` works, but I assume it just installs the grpc python plugin, which you can do separately and then call from `buf generate`